src/backend/distributed/commands/function.c

```diff
@@ -555,8 +563,94 @@ GetFunctionDDLCommand(const RegProcedure funcOid)
 	/* revert back to original search_path */
 	PopOverrideSearchPath();
 
-	sql = TextDatumGetCString(sqlTextDatum);
-	return sql;
+	createFunctionSQL = TextDatumGetCString(sqlTextDatum);
+	alterFunctionOwnerSQL = GetFunctionAlterUserCommand(funcOid);
+
+	appendStringInfo(ddlCommand, "%s;%s", createFunctionSQL, alterFunctionOwnerSQL);
```

Comment:
I think it always makes sense to append `alterFunctionOwnerSQL ` to `createFunctionSQL ` when `GetFunctionDDLCommand ` is called. Because we rely ob this on two other cases: (a) create or replace function (b) add node  In both cases, we should append the owner of the function.

Let mw know if you think otherwise